### DISCUSSION TITLE
PyMuPdf

### DISCUSSION THREAD

**Discussion by gitmatsika** [flet-dev/flet#3400] [Packages] [answered]
Add PyMupdf to flet
https://pymupdf.readthedocs.io/en/latest/about.html
https://pypi.org/project/PyMuPDF/
https://github.com/pymupdf/PyMuPDF/discussions/2680

**Comment by paul-upfeat** [comment]
+1 here, i know this is an old thread. I attempted to build the wheel for this but wasn't able to - it contains C++ and trying to patch it got a bit complex.

Much appreciated if possible!

**Comment by jeremyworger** [comment]
Any updates to adding this package?
I see there was https://github.com/flet-dev/mobile-forge/blob/main/recipes/pymupdf/test_pymupdf.py ?

**Comment by ndonkoHenri** [comment, marked as answer]
`pymupdf` 1.27.2.3 is now supported on both iOS and Android: https://pypi.flet.dev/pymupdf

- [Docs](https://github.com/flet-dev/mobile-forge/tree/main/recipes/pymupdf)
- [Example](https://github.com/flet-dev/mobile-forge/tree/main/recipes/pymupdf/examples/render-and-read)

Give it a try and let us know! 🤟🏿

**Reply by ndonkoHenri** [reply]
Oh btw, remember `pymupdf` is licensed AGPL: https://github.com/pymupdf/PyMuPDF/discussions/971 :)
